doc/gui/extension/example_library/front-end/src/BasicTable.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useMemo, useState } from "react";
+import {
+    createRequestDataUpdateAction,
+    useDispatch,
+    useDispatchRequestUpdateOnFirstRender,
+    useModule,
+    TaipyDynamicProps,
+    TableValueType,
+    RowType,
+    RowValue,
+} from "taipy-gui";
+
+interface BasicTableProps extends TaipyDynamicProps {
+    data: TableValueType;
+}
+
+const pageKey = "no-page-key";
+
+const BasicTable = (props: BasicTableProps) => {
+    const { data, updateVarName = "", updateVars = "", id } = props;
+    const [value, setValue] = useState<TableValueType>({});
```

Comment:
I'm not sure this is the right type...
As type of data = `Record<string, type of value>`
Then I guess, value's type would be `Record<string, Array<RowValue>>`